### DISCUSSION TITLE
Advice for using GitHub Pages

### DISCUSSION THREAD

**Discussion by johnkerl** [quarto-dev/quarto-cli#382] [Q&A] [answered]
Hi! I've been trying out Quarto for the last week or so and it's awesome. Several experiments working beautifully.

One experiment that isn't: I cannot get get Quarto to write to a directory that GitHub Pages recognizes.

I simply cannot reproduce https://quarto.org/docs/websites/publishing-websites.html#github-pages.

* https://github.com/johnkerl/quarto-ghp-experiment has a minimum reproducible example
* https://github.com/johnkerl/quarto-ghp-experiment/blob/main/README.md describes the failing setup

Thanks in advance for any and all advice!

**Comment by jjallaire** [comment, marked as answer]
The `output-dir` can only be a directory within the project dir. So they way to do this is to either:

1) Quarto project in the root with `output-dir: docs`
2) Quarto project in the `docs` directory with `output-dir: .`
3) Quarto project as you have it and then create a `post-render` script (https://quarto.org/docs/projects/quarto-projects.html#project-scripts) to copy or move the output to where you want it

**Reply by johnkerl** [reply]
This works -- thanks @jjallaire !! :)

1. Nice -- doesn't fit our use-case with code+docs but I'll do this for any docs-only repos we have -- thanks!
2. I'm using this successfully -- mixes `.html` artifacts with `.md` but this is OK
3. I really liked this idea, & tried it first -- however I found that `quarto preview` would then 404 :(

Really appreciate the feedback & super-excited about Quarto! :)

**Comment by jjallaire** [comment]
Yeah I agree that (2) isn't ideal but at least it works w/ your directory layout.

Roger the `quarto preview` problem (I hadn't thought of that). I think we will need a special feature/behavior that accounts for this scenario.
